src/container/Container.js

```diff
@@ -2,228 +2,347 @@ import { _, Promise } from '../globals';
 import ExtensionApi from './ExtensionApi';
 import { ServiceError } from './errors';
 
-function countOccurrences (array, item) {
-    return array.filter(_.isEqual.bind(_, item)).length;
+export default function Container (extensions, config) {
```

Comment:
Not sure I prefer the new style tbh. What was the problem with the previous one? (Aside from repetition between the lint and get (lint is still WIP and hasn't been tagged into a release yet, would've been on a branch but i've been working solo)).

More importantly `onGetComplete` is broken for the non-cached case as it waits for the promise resolution. This will break the no-cache extension at least.